test/test_filenames.py

```diff
@@ -0,0 +1,54 @@
+import os
+import pytest
+from pathlib import Path
+
+dir_list = [{
+    "type": "directory",
+    "name": "FYs",
+    "children": [
+        {
+        "type": "directory",
+        "name": "FY20",
+        "children": [
+            {
+            "type": "file",
+            "name": "letterofresignation_2020_10_04.docx"
+            },
+            {
+            "type": "file",
+            "name": "letterofresignation_2020_10_31.docx"
+            }
+        ]
+        },
+        {
+        "type": "file",
+        "name": "README.txt"
+        }
+    ]
+    }
+]
+
+def tree_builder(path, path_list):
+    for item in path_list:
+        if item["type"] == "file":
+            path.join(item["name"]).write('CONTENT')
+        elif item["type"] == "directory":
+            new_dir_path = path.mkdir(item["name"])
+            if "children" in item.keys() and isinstance(item["children"], list):
+                subdir = new_dir_path
+                recursive_tree_builder(subdir, item["children"])
+
+@pytest.fixture()
+def testing_directory(tmpdir):
+    tree_builder(tmpdir, dir_list)
```

Comment:
is this parametrisable? (i.e. is it possible to pass different dir_lists)

Reply:
That wouldn't work with current set up. Would be good to keep that in the backlog, great idea.

Reply:
I guess currently if dir_list is a global, it can be overwritten by a test during run time